### ISSUE TITLE
Add option to read image from standard in

### ISSUE THREAD

**Issue by mozirilla213** [karlch/vimiv-qt#459]
It would be great if there was a command line flag like `-` to read a raw image content from stdin and display it.

Use cases:
Making images with Imagemagick:
```
convert -pointsize 100 label:"text" bmp:- | vimiv -
```
Rendering pic pictures:
```
pic file.pic | pic2graph | vimiv -
```
Now the user wouldn't need to create a temporary file each time she wants to recompile the image

Similar applications like `zathura` have this option

Here's a very simple diff of commit 4a1a0d1 that does this by creating a file in the vimiv cache directory: http://ix.io/2NMG/, but I think a better approach should be made

**Comment by karlch** [comment]
I agree that this could be useful, thanks for the idea and the diff!

With "a better approach", do you mean avoiding a temporary file completely? I am not sure how well vimiv would deal with this, as especially the library very much relies on the concept of directories and files.

**Comment by crazyminecuber** [comment]
I think it would also be really useful to have a flag for printing the marked images on program exit. Then in combination with the input flag, you could use vimiv as visual dmenu filter. 

Usage: Load images from some number of folders, mark some of the images with the same theme, exit the application and pipe the resulting list of paths to another program that moves them to a certain directory or manipulates them in another way.

**Comment by karlch** [comment]
The "visual dmenu" concept should be possible, at least when reading paths from stdin, not binary images as proposed for this issue. There is the `-o` flag which allows printing arbitrary output which gets parsed just like the command line, e.g. `-o %m` will print marked paths upon exit. This was implemented in #381, but is currently not yet in any release in case you use the "stable" ("ancient", shame on us for our release schedule) version.

**Comment by mozirilla213** [comment]
Hi, sorry for the hiatus. I meant that if there's a different, more organized way vimiv or Qt libraries already handle cache files we should go for it (I haven't dive very deep in the source code). But if a simple temp file does the job, that's fine too. Also, I should have probably used the `xdg.vimiv_cache_dir()` directory in the diff.
